platform/platform_tests/apk_test.cpp

```diff
@@ -14,112 +15,113 @@
+  "languages.txt",
+  "unicode_blocks.txt",
+  "drules_proto_clear.bin",
+  "external_resources.txt",
+  "packed_polygons.bin",
+  "countries.txt"
+};
+
+class ApkTester : public threads::IRoutine
 {
-  char const * arrFiles[] = {
-    "copyright.html",
-    "resources-mdpi_clear/symbols.sdf",
-    "resources-mdpi_clear/symbols.png",
-    "resources-hdpi_clear/symbols.sdf",
-    "resources-hdpi_clear/symbols.png",
-    "resources-xhdpi_clear/symbols.sdf",
-    "resources-xhdpi_clear/symbols.png",
-    "categories.txt",
-    "categories_cuisines.txt",
-    "classificator.txt",
-    "types.txt",
-    "fonts_blacklist.txt",
-    "fonts_whitelist.txt",
-    "languages.txt",
-    "unicode_blocks.txt",
-    "drules_proto_clear.bin",
-    "external_resources.txt",
-    "packed_polygons.bin",
-    "countries.txt"
-  };
-
-  class ApkTester : public threads::IRoutine
+  static const int COUNT = ARRAY_SIZE(arrFiles);
+  string const & m_cont;
+
+public:
+  explicit ApkTester(string const & cont) : m_cont(cont), m_hashes(COUNT)
   {
-    static const int COUNT = ARRAY_SIZE(arrFiles);
-    string const & m_cont;
+  }
 
-  public:
-    explicit ApkTester(string const & cont) : m_cont(cont), m_hashes(COUNT)
-    {
-    }
+  virtual void Do()
+  {
+    string const prefix("assets/");
 
-    virtual void Do()
+    while (true)
     {
-      string const prefix("assets/");
+      size_t ind = rand() % COUNT;
+      if (m_hashes[ind] != 0)
+      {
+        ind = COUNT;
+        for (size_t i = 0; i < COUNT; ++i)
+          if (m_hashes[i] == 0)
+          {
+            ind = i;
+            break;
+          }
+      }
 
-      while (true)
+      if (ind == COUNT)
+        break;
+
+      try
       {
-        size_t ind = rand() % COUNT;
-        if (m_hashes[ind] != 0)
-        {
-          ind = COUNT;
-          for (size_t i = 0; i < COUNT; ++i)
-            if (m_hashes[i] == 0)
-            {
-              ind = i;
-              break;
-            }
-        }
-
-        if (ind == COUNT)
-          break;
-
-        try
-        {
-          ZipFileReader reader(m_cont, prefix + arrFiles[ind]);
-
-          size_t const size = reader.Size();
-          vector<char> buffer(size);
-          reader.Read(0, &buffer[0], size);
-
-          m_hashes[ind] = accumulate(buffer.begin(), buffer.end(), static_cast<uint64_t>(0));
-        }
-        catch (Reader::Exception const & ex)
-        {
-          LOG(LERROR, (ex.Msg()));
-          break;
-        }
+        ZipFileReader reader(m_cont, prefix + arrFiles[ind]);
+
+        size_t const size = reader.Size();
+        vector<char> buffer(size);
+        reader.Read(0, &buffer[0], size);
+
+        m_hashes[ind] = accumulate(buffer.begin(), buffer.end(), static_cast<uint64_t>(0));
+      }
+      catch (Reader::Exception const & ex)
+      {
+        LOG(LERROR, (ex.Msg()));
+        break;
       }
     }
+  }
 
-    vector<uint64_t> m_hashes;
-  };
-}
+  vector<uint64_t> m_hashes;
+};
 
+/*
```

Comment:
What's wrong with this test?

Reply:
It doesn't work without a valid apk. It would be better to enable it for Android builds and pass path to apk as a parameter.